### PR TITLE
Track test coverage via codecov

### PR DESCRIPTION
**Pull request by Martchus**
See https://progress.opensuse.org/issues/115022

### CODE REVIEW

**Comment by Martchus** [comment]
Looks like we're getting a coverage report now.

**Comment by Martchus** [comment]
Not sure why the codecov check doesn't show up anymore. It did before I force-pushed and I only added the batch. Maybe we can merge this nevertheless and see whether it works then. According to the logs the report is uploaded successfully to codecov and one can also see the coverage on codecov (https://codecov.io/gh/os-autoinst/os-autoinst-wheel-launcher/pull/3/tree).

**Review by kalikiana — APPROVED** [review]

**Comment by Martchus** [comment]
Oh, now the checks have appeared again. It seems that it just took a while (as I have done absolutely nothing on this branch anymore). So this should be definitely good to merge now (and how we structure the test execution once we add more tests is something to sort out in the next PR).

File: test [hunk 1]
```diff
@@ -1 +1,11 @@
+if [[ $WITH_COVER_OPTIONS ]]; then
+    cover_deps=()
+    perl -MDevel::Cover -e1 &> /dev/null || cover_deps+=('perl(Devel::Cover)')
+    perl -MDevel::Cover::Report::Codecovbash -e1 &> /dev/null || cover_deps+=('perl(Devel::Cover::Report::Codecovbash)')
+    [[ $cover_deps ]] && zypper --non-interactive in "${cover_deps[@]}"
```
Comment (perlpunk) [inline test:5]: Wouldn't it be enough to just do the zypper install, and it won't do anything if the modules are installed?

Comment (Martchus) [reply test:5]: I want to use this script also locally and then it is really inconvenient having to wait for zypper to refresh repos and run the resolver to find out that certain packages are already installed. Maybe there should be a different script for the CI and for local runs but for now I've just stuck with this approach.

**Pull request merged**